src/logger.js

```diff
@@ -0,0 +1,117 @@
+import { curry, compose } from 'ramda'
+import moment from 'moment'
+import { Console } from 'console'
+import _ from 'lodash'
+import fs from 'fs'
+
+const omitNil = curry(
+  (_, fn) => params => fn(_.omitBy(params, _.isNil))
+)(_)
+
+const filterLog = (
+  ({ LOGGING_LEVEL }) => {
+    const levelMap = {
+      emerg: 0,
+      alert: 1,
+      crit: 2,
+      err: 3,
+      warning: 4,
+      notice: 5,
+      info: 6,
+      debug: 7
+    }
+
+    return fn => params => {
+      const {
+        level
+      } = params
+
+      if (!LOGGING_LEVEL || levelMap[level] <= levelMap[LOGGING_LEVEL]) {
+        fn(params)
+      }
+    }
+  }
+)(process.env)
+
+const _logHandler = compose(
+  omitNil,
+  filterLog
+)
+
+const _logger = (
+  ({ LOGGING }, fs, Console, _logHandler) => {
+    let _console = console
+    if (LOGGING === 'file') {
+      if (!fs.existsSync('./output')) {
+        fs.mkdirSync('./output')
+      }
+      const output = fs.createWriteStream('./output/log.txt')
```

Comment:
I reckon this should either be an environment variable or somehow configured on the log beforehand

Reply:
Also I only found out today this should probably have the flags option passed so it appends to the same log.
`createWriteStream(fileName, { flags: 'a'})`